pkg/health/client/modules.go

```diff
@@ -56,3 +64,55 @@ func GetAndFormatModulesHealth(w io.Writer, clt ModulesHealth, verbose bool) {
 		tally[cell.StatusUnknown],
 	)
 }
+
+func addNode(p *node, path, meta string) {
+	// Skip daemon health for now...
+	if path == rootNode+".controlplane.daemon" {
+		return
+	}
+
+	parent, paths := p, strings.Split(path, ".")
+	for i, name := range paths {
+		name = strings.Replace(name, noPod, "", 1)
+		if i == len(paths)-1 {
+			parent.addNodeWithMeta(name, meta)
+			continue
+		}
+		if n := parent.find(name); n != nil {
+			parent = n
+			continue
+		}
+		if parent != nil {
+			parent = parent.addBranch(name)
+		} else {
+			parent = p.addBranch(name)
+		}
+	}
+}
+
+func buildTree(t *node, mid, s string) {
```

Comment:
FYI: With [#28773](https://github.com/cilium/cilium/pull/28773) I'm changing the way the module health report is sent - instead of sending a raw string a serialized version of the tree will be sent.  I think this is the way we want to do this going forward instead of trying to parse around raw formatted text output.

I'm a bit concerned about this approach since it seems brittle (i.e. it depends on an implicit formatting of text generated by cell), so changes there will need to reflected by changes here trying to parse out the text.

Alongside the changes in https://github.com/cilium/cilium/pull/28773, could we put all this logic in one place?

The text rendering code being in pkg/hive/cell is admittedly a bit awkward - perhaps we could do something like:

```go
type node struct {
    cell.StatusNode
    ...
}

func buildTree(t *node) { // module-id etc are no embedded in *node struct.  

}
```
So we can reuse some of that code and have all the code dealing rendering text live in the CLI code rather than in the hive/cell internals.

Wdyt?

Reply:
Thanks Tom! 
Right, I was trying to operate with what was there since that pr was not on the radar as of yet. I'll take a peek... 
I am actually not sure we want to send out a json version of the tree??
End of day, this data will likely be managed in statedb and queried from there. 
Hence my feel is we might be best served to keep sending out an array of module health reports.